File: my_py_pkg/my_py_pkg/add_two_ints_client_no_oop.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts
 
def main(args=None):
    rclpy.init(args=args)
    node = Node("add_two_int_client_no_oop") # MODIFY NAME

    client = node.create_client(AddTwoInts, "add_two_ints")
    while not client.wait_for_service(1.0):
        node.get_logger().warn("Waiting for Add Two Ints Server....")

    request = AddTwoInts.Request()
    request.a = 3 #number
    request.b = 7 #number2

    future = client.call_async(request)
    rclpy.spin_until_future_complete(node, future)

    response = future.result()
    
    node.get_logger().info(str(request.a) + "+" +
                        str(request.b) + "=" + str(response.sum))




    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/add_two_ints_client_no_oop.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"  

using namespace std::chrono_literals;

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("aad_two_ints_no_oop"); // MODIFY NAME
    
    auto client = node->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
    while (!client->wait_for_service(1s))
    {
        RCLCPP_WARN(node->get_logger(),"Waiting for the server ...");
    }
    
    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = 2;
    request->b = 5;
    
    auto future = client->async_send_request(request);
    rclcpp::spin_until_future_complete(node, future);

    auto response = future.get();
    RCLCPP_INFO(node->get_logger(),
                "%d + %d = %d", (int)request->a, (int)request->b, (int)response->sum);

    rclcpp::shutdown();
    return 0;
}